be/src/vec/sink/vrow_distribution.cpp

```diff
@@ -109,6 +111,54 @@ Status VRowDistribution::automatic_create_partition() {
     return status;
 }
 
+static TCreatePartitionResult cast_as_create_result(TReplacePartitionResult& arg) {
+    TCreatePartitionResult result;
+    result.status = arg.status;
+    result.nodes = std::move(arg.nodes);
+    result.partitions = std::move(arg.partitions);
+    result.tablets = std::move(arg.tablets);
+    return result;
+}
+
+// use _partitions and replace them
+Status VRowDistribution::_replace_overwriting_partition() {
+    SCOPED_TIMER(_add_partition_request_timer);
+    TReplacePartitionRequest request;
+    TReplacePartitionResult result;
+    request.__set_txn_id(_txn_id);
+    request.__set_db_id(_vpartition->db_id());
+    request.__set_table_id(_vpartition->table_id());
+
+    std::vector<int64_t> partition_ids(_partitions.size());
+    std::transform(_partitions.begin(), _partitions.end(), partition_ids.begin(),
+                   [](VOlapTablePartition* partition) { return partition->id; });
+    request.__set_partition_ids(partition_ids);
+
+    string be_endpoint = BackendOptions::get_be_endpoint();
+    request.__set_be_endpoint(be_endpoint);
+
+    LOG(WARNING) << "auto detect replace partition request: " << request;
+    TNetworkAddress master_addr = ExecEnv::GetInstance()->master_info()->network_address;
+    int time_out = _state->execution_timeout() * 1000;
+    RETURN_IF_ERROR(ThriftRpcHelper::rpc<FrontendServiceClient>(
+            master_addr.hostname, master_addr.port,
+            [&request, &result](FrontendServiceConnection& client) {
+                client->replacePartition(result, request);
+            },
+            time_out));
+
+    Status status(Status::create(result.status));
+    LOG(WARNING) << "auto detect replace partition result: " << result;
+    if (result.status.status_code == TStatusCode::OK) {
+        RETURN_IF_ERROR(_vpartition->replace_partitions(_partitions, result.partitions));
+        // reuse the function as the args' structure are same. it add nodes/locations and incremental_open
+        auto result_as_create = cast_as_create_result(result);
+        RETURN_IF_ERROR(_create_partition_callback(_caller, &result_as_create));
```

Comment:
where init `_create_partition_callback`, seems nullptr?

Reply:
in row_distribution's init. it reused auto_partition's same function.